programs/castle-vault/src/asset_container/rate.rs

```diff
@@ -40,55 +40,43 @@ mod tests {
 
     #[test]
     fn test_verify_weights_happy() {
-        let rates = AssetContainerGeneric::<Rate, 3> {
-            inner: [
-                Some(Rate::from_percent(0)),
-                Some(Rate::from_percent(0)),
-                Some(Rate::from_percent(100)),
-            ],
+        let rates = AssetContainerGeneric::<Rate, 2> {
+            inner: [Some(Rate::from_percent(0)), Some(Rate::from_percent(100))],
         };
         assert!(rates.verify_weights(100).is_ok())
     }
 
     #[test]
     fn test_verify_weights_happy2() {
-        let rates = AssetContainerGeneric::<Rate, 3> {
-            inner: [
-                Some(Rate::from_percent(1)),
-                Some(Rate::from_percent(59)),
-                Some(Rate::from_percent(40)),
-            ],
+        let rates = AssetContainerGeneric::<Rate, 2> {
+            inner: [Some(Rate::from_percent(59)), Some(Rate::from_percent(41))],
         };
         assert!(rates.verify_weights(59).is_ok())
     }
 
     #[test]
     fn test_verify_weights_unhappy_gt1() {
-        let rates = AssetContainerGeneric::<Rate, 3> {
-            inner: [
-                Some(Rate::from_percent(2)),
-                Some(Rate::from_percent(59)),
-                Some(Rate::from_percent(40)),
-            ],
+        let rates = AssetContainerGeneric::<Rate, 2> {
+            inner: [Some(Rate::from_percent(59)), Some(Rate::from_percent(42))],
         };
-        assert_eq!(
-            rates.verify_weights(100),
-            Err(ErrorCode::InvalidProposedWeights.into())
-        )
+        // TODO fix
```

Comment:
Is this a relevant TODO still? Can you fix and remove?

Also applies to L80

Reply:
yep, the previous `assert_eq` now gets me `binary operation == cannot be applied to type anchor_lang::error::Error`, so im not sure how to check for a specific ErrorCode. i think in this particular case its fine since there is only one possible error in the function it's testing, so we can just check if some error is thrown.